### PR TITLE
Filepath check update for noiseprof

### PR DESCRIPTION
**Pull request by Page-David**
Update `noiseprof` filepath check. The previous version did not consider `profile_path='noise.prof'` is legal.

### CODE REVIEW

**Comment by rabitt** [comment]
Hi @Page-David ,

I'm not sure I agree with adding the `format_output` function. Relative paths, for example `sox.file_info.duration('file_in_this_directory.wav')`, will already work. It looks like you want support for the `~` token in paths, which is specific to macs. The `macpath` library handles this problem , and I think the functionality should be separate from `pysox`.

**Comment by Page-David** [comment]
But I get this error in previous version.
```
Python 2.7.13 (default, Dec 21 2016, 07:16:46) 
[GCC 6.2.1 20160830] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import sox
>>> tfm = sox.Transformer()
>>> tfm.noiseprof('input.wav', 'noise.prof')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sox/transform.py", line 1881, in noiseprof
    raise IOError("profile_path {} is not writeable.".format(profile_path))
IOError: profile_path noise.prof is not writeable.
```

File: sox/transform.py [hunk 1]
```diff
@@ -1875,9 +1875,10 @@ def noiseprof(self, input_filepath, profile_path):
 
         '''
         if os.path.isdir(profile_path):
-            raise ValueError("profile_path {} is a directory, but should be a file")
+            raise ValueError("profile_path {} is a directory, but filename should be specified.")
         
-        if not os.access(os.path.dirname(profile_path), os.W_OK):
+        if (bool(os.path.dirname(profile_path)) and
```
Comment (rabitt) [inline sox/transform.py]: What about the case where the current directory is not writeable?

How about:
```python
if os.path.dirname(profile_path) == '':
    if not os.access(os.path.curdir):
        raise IOError("profile_path {} is not writeable.".format(os.path.curdir))
else:
    if not os.access(os.path.dirname(profile_path), os.W_OK):
        raise IOError("profile_path {} is not writeable.".format(profile_path))
```

**Comment by rabitt** [comment]
Hey @Page-David ah I see where you were running into problems. See my comment. Otherwise it's good to go.

File: sox/transform.py (hunk 1 above)
Comment (Page-David) [inline sox/transform.py]: `os.path.curdir` seems not working after `os.chdir` is called. and I found this [why-does-python-have-os-path-curdir](http://stackoverflow.com/questions/14512087/why-does-python-have-os-path-curdir)

**Pull request merged**